### PR TITLE
Use job_template_name as the TEST when scheduling jobs

### PR DESCRIPTION
**Pull request by Amrysliu**
Set the job_template_name as the TEST value if it exists.
This modification will cause those following changes:
```
  - apparmor_variant:
        testsuite: apparmor
        settings:
          START_AFTER_TEST: create_hdd_textmode
    - create_hdd_textmode_variant:
        testsuite: create_hdd_textmode
```
1. when users want to trigger `apparmor_variant` job with `TEST`
parameter, only `TEST=apparmor_variant` works. In this scenario, specify
the test suite name does not work.
2. When triggering jobs with `TEST=apparmor_variant`, the parent job
won't be triggered. If users want to both trigger the parent job, need
to define `START_AFTER_TEST: create_hdd_textmode_variant`.

Related pr: https://github.com/os-autoinst/openQA/pull/2553
See: https://progress.opensuse.org/issues/59097

### CODE REVIEW

File: t/api/02-iso.t [hunk 1]
```diff
@@ -1082,45 +1082,31 @@ subtest 'schedule tests correctly when changing TEST to job template name' => su
...
     $res = schedule_iso({%iso, _GROUP_ID => '1002', TEST => 'child_variant1', _SKIP_CHAINED_DEPS => 1});
-    is($res->json->{count}, 1, 'only schedule the child job with the same job template name');
-    my %child_job = map { $jobs->find($_)->TEST => 1 } @{$res->json->{ids}};
-    is_deeply(\%child_job, {child_variant1 => 1}, "only one child job was scheduled");
+    is($res->json->{count}, 1, 'do not schedule parent job');
+
+    add_opensuse_test('parallel_one', JOB_TEMPLATE_NAME => 'parallel_one_variant');
+    add_opensuse_test('parallel_two', PARALLEL_WITH     => 'parallel_one',);
+    $res = schedule_iso({%iso, _GROUP_ID => '1002', TEST => 'parallel_two'});
+    is($res->json->{count}, 0, 'could not find the parallel job');
```
Comment (okurz) [inline t/api/02-iso.t]: Don't we expect that at least "parallel_two" would be scheduled alone?

Comment (Amrysliu) [reply t/api/02-iso.t]: Actually `parallel_two` will be scheduled.  I check the code, in this situation, the value of `count` is 0 although the `parallel_two` already has been scheduled.

Comment (Amrysliu) [reply t/api/02-iso.t]: Actually parallel_two will be scheduled. I check the code, in this situation, the value of count is 0 although the parallel_two already has been scheduled.

Comment (Amrysliu) [reply t/api/02-iso.t]: Actually parallel_two will be scheduled. I check the code, in this situation, the value of count is 0 although the parallel_two already has been scheduled.
```
{
  count => 0,
  failed => [
    {
      error_messages => [
        "PARALLEL_WITH=selinux\@64bit not found - check for dependency typos and dependency cycles",
      ],
      job_id => 335,
    },
  ],
  ids => [],
  scheduled_product_id => 128,
}
```
The same result with `START_AFTER_TEST`.

Comment (okurz) [reply t/api/02-iso.t]: I don't understand. How did you check then that "parallel_two will be scheduled"?

Comment (Amrysliu) [reply t/api/02-iso.t]: I modified the code to check the failed message. The `job_id` is the job's id that is scheduled successfully.

Comment (okurz) [reply t/api/02-iso.t]: looks better. Thanks.

**Pull request merged**